build/llvm.ps1

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+. (Join-Path $PSScriptRoot "utils.ps1")
+
+# Build
+
+$srcRoot = Get-Location
+$buildDir = Join-Path $srcRoot build
+$llvmCmakeFile = Join-Path $buildDir llvm.cmake
+$llvmRootDir = Join-Path $srcRoot external llvm-project
+$llvmBuildDir = Join-Path $llvmRootDir build
+$llvmDir = Join-Path $llvmRootDir llvm
+
+if ($null -ne (get-command ccache -ErrorAction SilentlyContinue) ) {
+    Write-Vso "Found ccache command"
+
+    if (-not (Test-Path env:\CCACHE_DIR)) {
+        # CCACHE_DIR needs to be set, get the value from ccache
+        $Env:CCACHE_DIR = exec { ccache -k cache_dir }
+    }
+    Assert (![string]::IsNullOrWhiteSpace($Env:CCACHE_DIR)) "CCACHE_DIR is not set"
+
+    # Set cap and make sure dir is created
+    if(!(Test-Path $Env:CCACHE_DIR)) {
+        mkdir $Env:CCACHE_DIR | Out-Null
+    }
+    $Env:CCACHE_DIR = Resolve-Path $Env:CCACHE_DIR
+    ccache -M 2G
+
+    Write-Vso "Found ccache config:"
+    ccache --show-config
+
+    $cacheRoot = $Env:CCACHE_DIR
+
+    Assert (Test-Path $cacheRoot) "$($cacheRoot) is missing"
+    Write-Vso "Using CCACHE_DIR: $($Env:CCACHE_DIR)"
+}
+else {
+    Write-Vso "Did not find ccache command"
+}
+
+Assert (Test-Path $llvmDir) "llvm-project submodule is missing"
+Assert (![string]::IsNullOrWhiteSpace($Env:PKG_NAME)) "PKG_NAME is not set"
+
+Assert (Test-CommandExists "cmake") "CMAKE not found"
+Assert (Test-CommandExists "ninja") "Ninja-Build not found"
+if($IsLinux) {
+    Assert (Test-CommandExists "docker") "Docker not found"
+}
+
+if(!(Test-Path $llvmBuildDir)) {
+    mkdir $llvmBuildDir | Out-Null
+}
+
+Write-Vso "Generating package: $($Env:PKG_NAME)"
+
+if ($IsLinux) {
+    function Build-ContainerImage {
+
+        Write-Vso "Building container image manylinux-llvm-builder"
+        exec -wd $buildDir {
+            Get-Content manylinux.Dockerfile | docker build -t manylinux-llvm-builder --build-arg LLVM_BUILD_DIR="$llvmBuildDir" -
```

Comment:
Please keep args in a different line.

Reply:
What motivates piping `manylinux.Dockerfile` into `docker build` instead of using `docker build -f manylinux.Dockerfile`?